essentials/src/telemetry_subscription.rs

```diff
@@ -15,17 +15,50 @@
 // along with polkadot-introspector.  If not, see <http://www.gnu.org/licenses/>.
 //
 
-use crate::types::H256;
-
 use super::telemetry_feed::TelemetryFeed;
+use crate::{telemetry_feed::AddedChain, types::H256};
 use color_eyre::Report;
-use futures::SinkExt;
+use futures::{stream::Next, SinkExt};
 use futures_util::StreamExt;
+use itertools::Itertools;
 use log::{debug, info, warn};
 use priority_channel::{SendError, Sender};
+use std::{
+	cmp::{min, Reverse},
+	collections::HashMap,
+	io::{stdin, BufRead},
+};
 use tokio::{net::TcpStream, sync::broadcast::Sender as BroadcastSender};
 use tokio_tungstenite::{connect_async, tungstenite::Message, MaybeTlsStream, WebSocketStream};
-use url::Url;
+
+struct TelemetryStream {
+	ws_stream: WebSocketStream<MaybeTlsStream<TcpStream>>,
+}
+
+impl TelemetryStream {
+	async fn connect(url: &str) -> Self {
+		println!("Connecting to the telemetry server on {}\n", url);
+		let (ws_stream, _) = connect_async(url).await.expect("Failed to connect");
+		Self { ws_stream }
+	}
+
+	async fn subscribe_to(&mut self, chain: &H256) -> color_eyre::Result<()> {
+		match self.ws_stream.send(Message::Text(format!("subscribe:{:?}", chain))).await {
+			Ok(_) => {
+				info!("Subscribed to chain with hash {}", chain);
+				Ok(())
+			},
+			Err(e) => {
+				info!("Cannot subscribe to chain with hash {}: {:?}", chain, e);
+				Err(e.into())
+			},
+		}
+	}
+
+	fn next(&mut self) -> Next<'_, WebSocketStream<MaybeTlsStream<TcpStream>>> {
```

Comment:
I'm not sure it's idiomatic here

Reply:
It should be a part of another trait, specifically [Stream](https://docs.rs/futures/latest/futures/stream/trait.Stream.html) trait.

Reply:
So you need to implement `Stream` methods and that will allow to use `StreamExt` methods, such as `next`.